lib/common/cancel/src/blocking.rs

```diff
@@ -0,0 +1,35 @@
+use super::*;
+
+pub async fn on_drop<Out, Task>(task: Task) -> Result<Out, Error>
+where
+    Task: FnOnce(CancellationToken) -> Out + Send + 'static,
+    Out: Send + 'static,
+{
+    let cancel = CancellationToken::new();
+
+    let task = {
+        let cancel = cancel.child_token();
+        move || task(cancel)
+    };
+
+    let guard = cancel.drop_guard();
+    let output = tokio::task::spawn_blocking(task).await?;
+    guard.disarm();
+
+    Ok(output)
+}
+
+pub async fn on_token<Out, Task>(cancel: CancellationToken, task: Task) -> Result<Out, Error>
```

Comment:
Suggestion: rename to `cancelable_spawn_blocking_with_token`

Reply:
Done.